File: scenarios/command/dotnet/command_server/UnlockCommandServer.cs
using MQTTnet.Client;
using MQTTnet.Client.Extensions;
using proto_messages;

namespace command_server;

[RequestTopic("vehicles/{clientId}/command/{commandName}/request")]
public class UnlockCommandServer : CommandServer<UnlockRequest, UnlockResponse>
{
    public UnlockCommandServer(IMqttClient mqttClient)
        : base(mqttClient, "unlock", new ProtobufSerializer(UnlockRequest.Parser))
    {
    }
}

File: scenarios/command/ts/commandServer/src/unlockCommandServer.ts
import { MqttClient } from 'mqtt';
import { CommandServer } from '@mqttapplicationsamples/mqttjsclientextensions';
import {
    UnlockRequest,
    UnlockResponse,
    UnlockRequestSerializer,
    UnlockResponseSerializer
} from '@mqttapplicationsamples/protomessages';

export const UnlockCommand = 'unlock';
export const RequestTopicPattern = 'vehicles/{clientId}/command/{commandName}/request';

export class UnlockCommandServer extends CommandServer<UnlockRequest, UnlockResponse>
{
    constructor(mqttClient: MqttClient) {
        super(mqttClient, RequestTopicPattern, UnlockCommand, new UnlockRequestSerializer(), new UnlockResponseSerializer());
    }
}
